src/sage/modules/free_module.py

```diff
@@ -4966,27 +4966,103 @@ def subspace_with_basis(self, gens, check=True, already_echelonized=False):
         """
         return self.submodule_with_basis(gens, check=check, already_echelonized=already_echelonized)
 
-    def complement(self):
+    def complement(self, *, orthogonal=None):
         r"""
-        Return the complement of ``self`` in the
+        Return a complementary subspace of ``self`` in the
         :meth:`~sage.modules.free_module.FreeModule_ambient_field.ambient_vector_space`.
 
+        EXAMPLES::
+
+            sage: V = RR^3
+            sage: S = V.subspace([[1,0,0], [0,1,0]])
+            sage: C = S.complement(orthogonal=False)
+            sage: C
+            Vector space of degree 3 and dimension 1 over Real Field with 53
+            bits of precision
+            Basis matrix:
+            [0.000000000000000 0.000000000000000  1.00000000000000]
+            sage: C + S == V
+            True
+            sage: V = VectorSpace(GF(2), 3)
+            sage: S = V.subspace([[1,1,1], [0,0,1]])
+            sage: C = S.complement(orthogonal=False)
+            sage: C
+            Vector space of degree 3 and dimension 1 over Finite Field of size 2
+            Basis matrix:
+            [0 1 0]
+            sage: C + S == V
+            True
+            sage: C.complement(orthogonal=False) == S
+            False
+        """
+        if orthogonal is None:
+            from sage.misc.superseded import deprecation
+            deprecation(31487, "The functionality of the complement() function"
+                               + " of a vector space is being updated. The original"
+                               + " functionality is being moved to the"
+                               + " orthogonal_complement() function. This function"
+                               + " will instead return a complementary subspace")
+            return self.orthogonal_complement()
+        if orthogonal is True:
+            return self.orthogonal_complement()
+        if orthogonal is False:
```

Comment:
```suggestion
        else:
```

if users pass something that is none of True, False or None, they will have to deal with it themselves.